### ISSUE TITLE
Stop using prefix to define attributes

### ISSUE THREAD

**Issue by lmolkova** [open-telemetry/semantic-conventions#1292]
Attributes can be defined in two equivalent ways.

E.g. `foo.bar` can be defined with prefix `foo` and id `bar`

```yaml
group:
- id: foo
  prefix: foo
  attributes: 
  - id: bar
```

or

without prefix and id `foo.bar`

```yaml
group:
- id: foo
  attributes: 
  - id: foo.bar
```

Using prefix is inconvenient:
- when searching for attribute definition in yaml files
- when reading yaml definitons, it's not obvious what the full attribute name would be generated
- tooling needs to resolve full attribute name and we need to be aware of it in some cases when writing policy checks, writing jq filters, generating code

Removing prefix would keep all MD files intact, but would resolve these inconveniences without any obvious downsides.

**Comment by lmolkova** [comment]
Another option: prefix is only used to validate that attributes start with it

```yaml
group:
- id: foo
  prefix: foo
  attributes: 
  - id: foo.bar
```

This allows to embedding #1264 to work and keeps attributes in the same group consistent, but also solves all the problems with partial name in the `id`.

**Comment by lmolkova** [comment]
The canonical name is globally unique today (within otel semconv) and our tooling tests for it. Do I miss something?

**Comment by lmolkova** [comment]
I believe it's fixed with #1293.

#1292 stays open to track schema changes
